helpers/templates/humamize_duration_test.go

```diff
@@ -0,0 +1,80 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package templates
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestHumanizeDurationSecondsFloat64(t *testing.T) {
+	input := []float64{0, 1, 60, 3600, 86400, 86400 + 3600, -(86400*2 + 3600*3 + 60*4 + 5), 899.99}
+	expected := []string{"0s", "1s", "1m 0s", "1h 0m 0s", "1d 0h 0m 0s", "1d 1h 0m 0s", "-2d 3h 4m 5s", "14m 59s"}
+
+	for index, value := range input {
+		result, err := HumanizeDuration(value)
+		require.NoError(t, err)
+		require.Equal(t, expected[index], result)
+	}
```

Comment:
nit: The test themselves are lovely, great cases.

It's more idiomatic in go (and even better for IDEs) if we structure the tests as a table. As such, I think a better way to structure your tests is:

```golang
	tc := []struct {
		name     string
		input    float64
		expected string
	}{
		{name: "zero", input: 0, expected: "0s"},
		{name: "one second", input: 1, expected: "1s"},
		{name: "one minute", input: 60, expected: "1m 0s"},
		{name: "one hour", input: 3600, expected: "1h 0m 0s"},
		{name: "one day", input: 86400, expected: "1d 0h 0m 0s"},
		{name: "one day and one hour", input: 86400 + 3600, expected: "1d 1h 0m 0s"},
		{name: "negative duration", input: -(86400*2 + 3600*3 + 60*4 + 5), expected: "-2d 3h 4m 5s"},
		{name: "Using a float", input: 899.99, expected: "14m 59s"},
	}
	
	for _, tt := range tc {
		t.Run(tt.name, func(t *testing.T) {
			result, err := HumanizeDuration(tt.input)
			require.NoError(t, err)
			require.Equal(t, tt.expected, result)
		},
	}

```

The same comment applies for all of the tests in the file.